File: Add Code Here/JAVA/Sorting Leetcode Questions/MergeSortedArray.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
         int i1 = m - 1;  
         int i2 = n - 1; 
         
         
         while (i1 >= 0 && i2 >= 0) {
             if (nums1[i1] >= nums2[i2]) {
                 nums1[i1 + i2 + 1] = nums1[i1--];
             } else {
                 nums1[i1 + i2 + 1] = nums2[i2--];
             }
         }
         
         while (i2 >= 0) {
             nums1[i2] = nums2[i2--];
         }
     }
 }

File: Add Code Here/C++/tree/C++/Merge Sorted Array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int i =m-1, j= n-1, index = m+n -1;
        while(i>=0 && j>=0 )
        {
            if(nums1[i] >= nums2[j]){
                nums1[index--] = nums1[i--];
            }
            else{
                nums1[index--]= nums2[j--];
            }
        }
        
        while(j>=0){
            nums1[index--] = nums2[j--];
        }
        
    }
};